sanity-check/main.py

```diff
@@ -0,0 +1,214 @@
+        return_metadata['tumour_normal_designation']=response.json()['specimens'][specimen_ind[0]]['tumourNormalDesignation']
+        return_metadata['specimen_type']=response.json()['specimens'][specimen_ind[0]]['specimenType']
+        return_metadata['submitter_specimen_id']=response.json()['specimens'][specimen_ind[0]]['submitterId']
+
+        sample_ind=[ele for ele,sample in enumerate(response.json()['specimens'][specimen_ind[0]]['samples']) if sample['sampleId']==return_metadata['sample_id']]
+        if len(sample_ind)!=1:
+            sys.exit("ID Mismatch detected. Sample_id:'%s'/'%s'  was not found within Specimen:'%s'/'%s' 's samples" % (metadata['submitter_sample_id'],return_metadata['sample_id'],metadata['submitter_specimen_id'],return_metadata['specimen_id'],))
+        return_metadata['sample_type']=response.json()['specimens'][specimen_ind[0]]['samples'][sample_ind[0]]['sampleType']
+        return_metadata['submitter_sample_id']=response.json()['specimens'][specimen_ind[0]]['samples'][sample_ind[0]]['submitterId']
+
+    return return_metadata
+
+def compile_metadata(metadata,clinical_metadata):
+    
+    ###Over-write metadata with clinical_metadata
+    return_metadata=metadata.copy()
+    for key in clinical_metadata.keys():
+        return_metadata[key]=clinical_metadata[key]
+    return return_metadata
+
+def check_study_exists(metadata,submission_song_url):
+    headers={"accept":"*/*"}
+    endpoint="%s/studies/%s" % (submission_song_url,metadata['program_id'])
+    response=requests.get(endpoint,headers=headers)
+    
+    if response.status_code==404:
+        sys.exit("Program %s does not exist. Please verify program code is correct. Otherwise contact DCC-admin for help to troubleshoot." % (metadata.get('program_id')))
+    elif response.status_code!=200:
+        sys.exit("Unable to fetch. Status code : %s" % (str(response.status_code)))
+    else:
+        return True
+                                
+def check_analysis_exists(metadata,submission_song_url):
+    headers={"accept":"*/*"}
+    endpoint="%s/studies/%s/analysis/search/id?sampleId=%s" % (submission_song_url,metadata.get('program_id'),metadata.get('sample_id'))
+
+    response=requests.get(endpoint,headers=headers)
+    
+    if response.status_code!=200:
+        sys.exit("Unable to fetch. Status code : %s" % (str(response.status_code)))
+    elif response.status_code==200:
+        ### If first submission of sample, query will return empty list
+        if len(response.json())==0:
+            return True
+        else:
+        ### Handles instances where sample has existing analyses
+            for analysis in response.json():
+                ### If analysis is suppressed; we ignore
+                if analysis["analysisState"]=="PUBLISHED" and analysis["experiment"]["experimental_strategy"]==metadata.get('experimental_strategy'):
```

Comment:
Need to add `analysisType=="sequencing_experiment"`

Reply:
Updated:
https://github.com/icgc-argo/argo-data-submission/blob/2a6ee2ba6bf1716bd58ed3d3e008bc08594d0e9c/sanity-check/main.py#L190-L194